### PR TITLE
Add logging to `download:plugins` script

### PR DESCRIPTION
**Pull request by tsmaeder**
#### What it does
Adds logging to get more info on CI failures descibed in #13902

Contributed on behalf of STMicroelectronics

#### How to test
Rerun the CI scripts until they fail.

#### Review checklist

- [ ] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: dev-packages/ovsx-client/src/ovsx-http-client.ts [hunk 1]
```diff
@@ -34,28 +34,12 @@ export class OVSXHttpClient implements OVSXClient {
...
-            return await this.requestJson(this.buildUrl('api/-/search', searchOptions));
-        } catch (err) {
-            return {
-                error: err?.message || String(err),
-                offset: -1,
-                extensions: []
-            };
-        }
+    search(searchOptions?: VSXSearchOptions): Promise<VSXSearchResult> {
+        return this.requestJson(this.buildUrl('api/-/search', searchOptions));
```
Comment (msujew) [inline dev-packages/ovsx-client/src/ovsx-http-client.ts:38]: I believe this API has a consumer on the `browser` part of `@theia/vsx-registry`. I think it relies on `VSXSearchResult#error` being set - just throwing will lead to issues in the extension search widget.

File: dev-packages/ovsx-client/src/ovsx-http-client.ts [hunk 2]
```diff
@@ -34,28 +34,12 @@ export class OVSXHttpClient implements OVSXClient {
...
-            return await this.requestJson(this.buildUrl('api/v2/-/query', queryOptions));
-        } catch (error) {
-            return {
-                offset: 0,
-                totalSize: 0,
-                extensions: []
-            };
-        }
+    query(queryOptions?: VSXQueryOptions): Promise<VSXQueryResult> {
+        return this.requestJson(this.buildUrl('api/v2/-/query', queryOptions));
```
Comment (msujew) [inline dev-packages/ovsx-client/src/ovsx-http-client.ts:42]: Similar to the issue above.

File: dev-packages/ovsx-client/src/ovsx-http-client.ts (hunk 1 above)
Comment (tsmaeder) [reply dev-packages/ovsx-client/src/ovsx-http-client.ts:38]: Yes, I know...we'll cross that bridge when we get to it.

Comment (msujew) [reply dev-packages/ovsx-client/src/ovsx-http-client.ts:38]: Ok, sure 👍

**Review by msujew — APPROVED** [review]

File: packages/vsx-registry/src/browser/vsx-extensions-model.ts [hunk 3]
```diff
@@ -225,49 +225,59 @@ export class VSXExtensionsModel {
...
+                                    verified: verified
+                                }));
+                                return Promise.resolve();
+                            });
                         });
-                    });
+                    }
                 }
+            } catch (error) {
+                this._searchError = error;
```
Comment (msujew) [inline packages/vsx-registry/src/browser/vsx-extensions-model.ts:262]: Suggestion: We should stringify this error somehow, as `_searchError` is defined as a string. (while `error` is `any` and likely of class `Error`)

**Review by msujew — APPROVED** [review]
Looks good, just one suggestion, see below.

File: packages/vsx-registry/src/browser/vsx-extensions-model.ts (hunk 3 above)
Comment (tsmaeder) [reply packages/vsx-registry/src/browser/vsx-extensions-model.ts:262]: The equivalent of what we had before would be `err?.message || String(err),`.

**Pull request merged**